Sources/BranchSDK/Branch.m

```diff
@@ -1922,11 +1923,21 @@ - (void)initSafetyCheck {
 
 - (void)initUserSessionAndCallCallback:(BOOL)callCallback sceneIdentifier:(NSString *)sceneIdentifier urlString:(NSString *)urlString {
     
-    // ignore lifecycle calls while waiting for a plugin runtime.
     @synchronized (self) {
         if (self.deferInitForPluginRuntime) {
-            [[BranchLogger shared] logDebug:@"Branch init is deferred, ignoring init call."];
+            if (urlString) {
+                [[BranchLogger shared] logDebug:@"Branch init is deferred, caching link"];
+                self.cachedURLString = urlString;
+            } else {
+                [[BranchLogger shared] logDebug:@"Branch init is deferred, ignoring lifecycle call without a link"];
+            }
             return;
+        } else {
+            if (!urlString && self.cachedURLString) {
+                urlString = self.cachedURLString;
+                self.cachedURLString = nil;
```

Comment:
Move this line out of if so that the cached url will get cleared, in case of race with new url after deferred init is unlocked